lib/racecar/config.rb

```diff
@@ -250,6 +256,35 @@ def rdkafka_producer
       producer_config
     end
 
+    def instrumenter
+      @instrumenter ||= begin
+        default_payload = { client_id: client_id, group_id: group_id }
+
+        if defined?(ActiveSupport::Notifications)
+          # ActiveSupport needs `concurrent-ruby` but doesn't `require` it.
+          require 'concurrent/utility/monotonic_time'
+          Racecar::Instrumenter.new(backend: ActiveSupport::Notifications, default_payload: default_payload)
+        else
+          logger.warn "ActiveSupport::Notifications not available, instrumentation is disabled"
+          NullInstrumenter
+        end
+      end
+    end
+    attr_writer :instrumenter
+
+    def install_liveness_probe
+      liveness_probe.tap(&:install)
+    end
+
+    def liveness_probe
+      require "active_support/notifications"
+      @liveness_probe ||= LivenessProbe.new(
+        ActiveSupport::Notifications,
```

Comment:
Should this be `instrumenter`?

Reply:
`LivenessProbe` needs to subscribe to the instrumentation events and the `Racecar::Instrumenter` class only implements `#instrument`.

The other reason it's not `instumenter` is that we have this optional `ActiveSupport::Notifications` dependency that is actually required for the liveness probe.

I would be super happy to have the flexibility of passing in your own instrumentation backend but I'm not sure anyone wants that.